torch/_dynamo/variables/torch.py

```diff
@@ -458,6 +458,29 @@ def get_state_from_generator():
                     if isinstance(x.value, np.generic):
                         x.value = x.value.item()
 
+            if self.value == torch._C._nn.scaled_dot_product_attention:
+                # We need to check if flash_attention will run and if dropout is != 0.0
+                # if that is the case then we want dynamo to graph break because this is
+                # not function correctly with cuda graphs and fake_tensor()
+                # TODO `get_fake_value(args[0].as_proxy().node, tx)[0]` was stripping the
+                # first dim from size from the fake tensor.. why?
+                fake_query = args[0].as_proxy().node.meta["example_value"]
+                fake_key = args[1].as_proxy().node.meta["example_value"]
+                fake_value = args[2].as_proxy().node.meta["example_value"]
+                backend_choice = torch._fused_sdp_choice(
```

Comment:
This is failing on one amp test. I think its because AMP hasn't desugred the dtype to its lower precision format for the fake tensor. Hence backend_choice is not flash_attention but at runtime it is one of the alllowed dtypes and we head the stream check in the kernel.

Reply:
torch.is_autocast_enabled() is returning false when it should be returning true?

Reply:
If I wrap the function that calls dynamo.optomize() with : `@torch.autocast("cuda")` then `torch.is_autocast_enabled()` does return true

Reply:
Offline discussed with @drisspg, we can use ```tx.block_stack[-1].with_context``` to get the ```AutocastModeVariable``` and then get the autocast state from there.